### PR TITLE
API endpoint for listing feedbacks

### PR DESCRIPTION
**Pull request by zetmann**
Solves [RC-146](https://dbmedialab.atlassian.net/browse/RC-146)

### CODE REVIEW

File: src/app/db/schemas/ArticleSchema.ts [hunk 1]
```diff
@@ -42,6 +42,9 @@ const ArticleSchema : Schema = new Schema({
 	toObject: {
 		retainKeyOrder: true,
 		transform: (doc : Document, converted : any) => {
+			// console.log('------------------------------------------------------------');
+			// console.log('ArticleSchema transform:', converted);
```
Comment (GrimStal) [inline src/app/db/schemas/ArticleSchema.ts]: And this

Comment (phgroe) [inline src/app/db/schemas/ArticleSchema.ts]: These are my bad, I left them in for easy debugging of additional query functions 😊

File: src/app/services/feedback/FeedbackDAO.ts [hunk 2]
```diff
@@ -39,32 +46,92 @@ import emptyCheck from 'app/util/emptyCheck';
...
 	article : Article,
 	enduser : EndUser,
 	items : FeedbackItem[]
-) : Promise <Feedback> {
+) : Promise <Feedback>
+{
 	emptyCheck(article, enduser, items);
 
+	// console.log('------------------------------------------------------------');
+	// console.log('article object in feedback.save:', article);
```
Comment (GrimStal) [inline src/app/services/feedback/FeedbackDAO.ts]: And this

File: src/app/services/feedback/common/validateAndSave.ts [hunk 3]
```diff
@@ -48,13 +48,18 @@ export default function(data : any) : Promise <Feedback> {
...
 	let article : Article;
-	let user : EndUser;
+	let enduser : EndUser;
 
 	return Promise.all([
-		getArticle(data.article).then((a : Article) => article = a),
-		getEndUser(data.user).then((u : EndUser) => user = u),
+		getArticle(data.article).then((a : Article) => {
+			// console.log('------------------------------------------------------------');
+			// console.log('validateAndSave got article:', a);
```
Comment (GrimStal) [inline src/app/services/feedback/common/validateAndSave.ts]: This is too

**Comment by phgroe** [comment]
I'm merging my stuff from #249 and #258 into develop now, since it is working against this branch. Then we get a clean history here.

File: src/app/routes/admin/api/feedbacksHandler.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { feedbackService } from 'app/services';
+
+/**
+ * Provides with whole list of existing feedbacks
+ * Not filtering, no page or limit query params are taken into account
+ */
+export function list (requ: Request, resp: Response) {
+	const notFound = 'Resourse not found';
+	feedbackService.getRange().then((fbacks) => {
+		if (fbacks.length > 0) {
```
Comment (phgroe) [inline src/app/routes/admin/api/feedbacksHandler.ts]: Just a note for the future, can we simplify this code even more? A check against result size will apply to pretty much all bulk queries. This if/else can be easily put into a function (which might belong into the `apiResponse` module) and then calling the service query function + delivering the response will become nice and short:
```
feedbackService.getRange()
.then(feedbacks => bulkResponse(resp, feedbacks))
.catch(error => errorResponse(resp, error));
```
Not for this branch, but probably later...

Also there should be a common function to extract skip, limit and sort parameters from the request data. These parameters are currently missing here and should be added soon. But now let's get this delivered.

**Pull request merged**